From 19f18ca7fa7ae9a574b5ee346150f4b8abed5dc8 Mon Sep 17 00:00:00 2001
From: Haoran Wang <b50027@freescale.com>
Date: Thu, 12 Feb 2015 16:07:08 +0800
Subject: [PATCH 11/22] MA-6230 Fix hostapd authentication issue on Atheros
 wifi

Apply patch from Silex to fix the authentication issue.

Signed-off-by: Haoran Wang <b50027@freescale.com>
---
 src/ap/ap_config.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/ap/ap_config.c b/src/ap/ap_config.c
index 455013a..541f1d0 100644
--- a/src/ap/ap_config.c
+++ b/src/ap/ap_config.c
@@ -1,6 +1,7 @@
 /*
  * hostapd / Configuration helper functions
  * Copyright (c) 2003-2014, Jouni Malinen <j@w1.fi>
+ * Copyright (C) 2015 Freescale Semiconductor, Inc.
  *
  * This software may be distributed under the terms of the BSD license.
  * See README for more details.
@@ -920,6 +921,10 @@ void hostapd_set_security_params(struct hostapd_bss_config *bss,
 
 	if ((bss->wpa & 2) && bss->rsn_pairwise == 0)
 		bss->rsn_pairwise = bss->wpa_pairwise;
+#ifndef REALTEK_WIFI_VENDOR
+    else if (bss->wpa == 2)
+        bss->wpa_pairwise = bss->rsn_pairwise;
+#endif
 	bss->wpa_group = wpa_select_ap_group_cipher(bss->wpa, bss->wpa_pairwise,
 						    bss->rsn_pairwise);
 
-- 
1.8.0

